External/NIfTI_20140122/save_nii_v2.m

```diff
@@ -52,11 +52,15 @@ function save_nii_v2(nii, fileprefix, old_nii_fname,datatype)
 if ~isstruct(nii)
     if exist('old_nii_fname','var')
         img = nii;
-        nii = load_nii(old_nii_fname,[],[],[],[],[],1);
+        if ischar(old_nii_fname)
+            nii = load_nii(old_nii_fname,[],[],[],[],[],1);
+        elseif isstruct(old_nii_fname)
+            nii = old_nii_fname;
+        end
         for it=1:size(img,4)
             tmp(:,:,:,it) = unxform_nii(nii, img(:,:,:,it));
         end
-        if size(unxform_nii(nii, nii.img(:,:,:,1))) ~= size(unxform_nii(nii, img(:,:,:,1))), error('old_nii_fname doesn''t have the same dimension as your new data'); end
+        if isfield(nii,'img') && size(unxform_nii(nii, nii.img(:,:,:,1))) ~= size(unxform_nii(nii, img(:,:,:,1))), error('old_nii_fname doesn''t have the same dimension as your new data'); end
```

Comment:
Sorry for the late review, I thought my comment was submitted, but found out through Ilana since she couldn't see my comment that submitting reviews is a two-step process.

@tanguyduval why was the `isfield(nii, 'img')` condition added?

Reply:
@mathieuboudreau nice catch. sorry for this. I'm correcting this directly on Master.

I changed the behavior when saving. 

Before, the header was extracted from the input file during the save. This could be problematic if the user was moving the file before the fitting was finished (note that fitting can last a day!).

Now, the header is also stored in the appdata and is used when saving:
https://github.com/neuropoly/qMRLab/pull/193/files#diff-fd685c27cea515ed544adc058b390beeR401

Reply:
Ah ok, thanks @tanguyduval ! No worries!